api/src/main/java/org/commonjava/indy/data/DefaultStoreValidator.java

```diff
@@ -0,0 +1,173 @@
+package org.commonjava.indy.data;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpHead;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.commonjava.cdi.util.weft.ExecutorConfig;
+import org.commonjava.cdi.util.weft.WeftManaged;
+import org.commonjava.indy.IndyWorkflowException;
+import org.commonjava.indy.conf.IndyConfiguration;
+import org.commonjava.indy.model.core.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+
+public class DefaultStoreValidator implements StoreValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultStoreValidator.class);
+
+    @Inject
+    @WeftManaged
+    @ExecutorConfig( named="store-validation", threads=2, priority=6 )
+    private ExecutorService executorService;
+
+    @Inject
+    IndyConfiguration configuration;
+
+    @Override
+    public ArtifactStoreValidateData validate(ArtifactStore artifactStore) throws InvalidArtifactStoreException, MalformedURLException {
+        final CountDownLatch httpRequestsLatch = new CountDownLatch(2);
+        final HashMap<String, String> errors = new HashMap<>();
+        Optional<URL> remoteUrl = Optional.empty();
+
+        try {
+            if(StoreType.remote == artifactStore.getType()) {
+                // Cast to Remote Repository
+                RemoteRepository remoteRepository = (RemoteRepository) artifactStore;
+                boolean disabled = remoteRepository.isDisabled();
+                if(disabled) {
```

Comment:
Sorry, I didn't catch this earlier. For any ArtifactStore variant, if it's disabled, we should return a successful validation result. 

It's a normal situation that the repo would be disabled for various reasons (could even be that there are network problems between Indy and the server). That means we should try not to throw exceptions if we encounter them...our logic should accommodate it.

I think we should pass them through without applying any validation logic, since they're inert in the system. When we re-enable them, that will involve storing the repository again, and it should be re-validated here.

Reply:
I have do this , you can check it out @jdcasey